storybookjs/storybook
# Webpack: Make manager and preview build processes cancelable

Issue: N/A

## What I did

Transformed the build and start steps, for both manager and preview, for both webpack4 and webpack5, so that they are all generators. This allows us to e.g. cancel the execution of preview when manager fail and vice versa. 
The reason we need this is because the processes happen in parallel, so we need a generator in order to get some control over their executions.

There are lots of duplication between the builders, they certainly can be improved.

co-authored by @ndelangen 

## How to test

https://user-images.githubusercontent.com/1671563/160439661-4ee02178-b7c9-4a17-b7ed-09fe2b1c1e21.mp4

Sample code from video:
```js
// lib/core-server/build-static.ts | lib/core-server/build-dev.ts
await prompts({
      type: 'confirm',
      name: 'test',
      message: `Do you want to answer?`,
});
```

```js
// examples/react-ts/main.ts

const throwError = async () => {
  throw new Error('bla');
};

const previewWebpack = true;

if (!previewWebpack) {
  config.webpackFinal = throwError;
} else {
  // @ts-ignore
  config.managerWebpack = throwError;
}
```




If your answer is yes to any of these, please make sure to include it in your PR.
